src/Spago/Config.js

```diff
@@ -99,3 +99,25 @@ export function addRangesToConfigImpl(doc, rangesMap) {
 export function setPackageSetVersionInConfigImpl(doc, version) {
   doc.setIn(["workspace", "package_set", "registry"], version);
 }
+
+export function migrateV1ConfigImpl(doc) {
+  // see here for more info about the visitor: https://eemeli.org/yaml/#finding-and-modifying-nodes
+  let hasChanged = false;
+  Yaml.visit(doc, {
+    Pair(_idx, pair, _path) {
+      // A previous version of the config used underscores in the config keys,
+      // but we standardised on camelCase instead.
+      // So here we crawl through the keys of each YamlMap and convert all the keys
+      // to camelCase.
+      if (pair.key && Yaml.isScalar(pair.key)) {
+        pair.key.value = pair.key.value.replaceAll(/_./g, function (match) {
+          hasChanged = true;
+          return match.charAt(1).toUpperCase();
+        });
+      }
+    }
+  });
+  if (hasChanged) {
+    return doc;
+  }
+}
```

Comment:
So, this effectively disables snake_case entirely from the config forever. Moreover, even if someone decided to add a superfluous key to the YAML file in that particular case convention, this script will force it to be camel-case, even if that's not what the user wants. I think that's fine.

Reply:
> if someone decided to add a superfluous key to the YAML file in that particular case convention, this script will force it to be camel-case, even if that's not what the user wants

Once we find a way to go at #1165 then I think we could effectively disallow superfluous fields, they only seem to cause confusion (see the comment I left in the Config module)